src/api/skygear.ts

```diff
@@ -15,24 +14,22 @@ function defaultHeaders(context: CLIContext) {
 
 // tslint:disable-next-line:no-any
 export async function handleFailureResponse(response: Response): Promise<any> {
-  const payload = await response
-    .json()
-    .then((p) => {
-      return p;
-    })
-    .catch((_error) => {
-      throw new Error(response.statusText);
-    });
+  // By default, we use statusText as error message
+  // This is a fallback only.
+  const preparedError = new Error(response.statusText);
+  // Attach response to error
+  (preparedError as any).response = response;
```

Comment:
Should we have a subclass of `Error` for controller network errors? It'll avoid the need of duck typing to handle our errors.

Reply:
Let me introduce a `HTTPError`